codepropertygraph/src/main/scala/io/shiftleft/passes/ParallelCpgPass.scala

```diff
@@ -220,15 +217,15 @@ abstract class ConcurrentWriterCpgPass[T <: AnyRef](cpg: Cpg, outName: String =
       } finally {
         try {
           writer.queue.put(None)
-          writerThread.join()
+          Await.ready(task, Duration.Inf)
```

Comment:
We should probably have some reasonable timeouts but that's beyond the scope of this change since previously essentially it was doing the same thing

Reply:
Yes and agreed.